### PR TITLE
Use the new Riemann client in the Riemann sink

### PR DESCRIPTION
**Pull request by mcorbin**
The Riemann sink now uses https://github.com/riemann/riemann-go-client
instead of Goryman.

The sink now sends events to Riemann using TCP. It also adds a
default tag (`heapster`) to events (to easily filter events from Heapster in Riemann). 

The new sink sends batch of events to Riemann instead of sending them one by one. The batch size can be
controlled with a new parameter called "batchsize".

I have updated the documentation and the test suite.

cc @jamtur01 and @jsoriano for review

### CODE REVIEW

File: metrics/sinks/riemann/driver_test.go [hunk 1]
```diff
@@ -175,23 +247,145 @@ func TestStoreMultipleDataInput(t *testing.T) {
...
+				Value: pb.String("aaaa-bbbb-cccc-dddd"),
+			},
+		},
+	}
+	var expectedEvents = [5]*proto.Event{
+		event1,
+		event2,
+		event3,
+		event4,
+		event5,
```
Comment (jsoriano) [inline metrics/sinks/riemann/driver_test.go]: These events could be defined here inline

File: metrics/sinks/riemann/driver_test.go [hunk 2]
```diff
@@ -175,23 +247,145 @@ func TestStoreMultipleDataInput(t *testing.T) {
...
+				Key:   pb.String("namespace_id"),
+				Value: pb.String("123"),
+			},
+			{
+				Key:   pb.String("pod_id"),
+				Value: pb.String("aaaa-bbbb-cccc-dddd"),
+			},
+		},
+	}
+	var expectedEvents = [5]*proto.Event{
```
Comment (jsoriano) [inline metrics/sinks/riemann/driver_test.go]: No need to fix array size here

File: metrics/sinks/riemann/driver.go [hunk 3]
```diff
@@ -115,52 +121,53 @@ func (sink *riemannSink) Stop() {
...
-	}
+	var events []riemanngo.Event
 
 	for _, metricSet := range dataBatch.MetricSets {
 		host := metricSet.Labels[core.LabelHostname.Key]
 		for metricName, metricValue := range metricSet.MetricValues {
 			if value := metricValue.GetValue(); value != nil {
-				appendMetric(host, metricName, riemannValue(value), metricSet.Labels)
+				timestamp := dataBatch.Timestamp.Unix()
+				// creates and event and add it to dataEvent
```
Comment (mcorbin) [inline metrics/sinks/riemann/driver.go]: Fixed, thanks

File: metrics/sinks/riemann/driver.go [hunk 4]
```diff
@@ -172,42 +179,33 @@ func (sink *riemannSink) ExportData(dataBatch *core.DataBatch) {
 				for k, v := range metric.Labels {
 					labels[k] = v
 				}
-				appendMetric(host, metric.Name, riemannValue(value), labels)
+				timestamp := dataBatch.Timestamp.Unix()
+				// creates and event and add it to dataEvent
```
Comment (mcorbin) [inline metrics/sinks/riemann/driver.go]: Fixed, thanks

File: metrics/sinks/riemann/driver.go [hunk 5]
```diff
@@ -172,42 +179,33 @@ func (sink *riemannSink) ExportData(dataBatch *core.DataBatch) {
...
+				appendEvent(events, sink, host, metric.Name, value, labels, timestamp)
 			}
 		}
 	}
-
-	if len(dataEvents) > 0 {
-		sink.sendData(dataEvents)
+	// Send events to Riemann if dataEvents is not empty
```
Comment (mcorbin) [inline metrics/sinks/riemann/driver.go]: Fixed, thanks

File: metrics/sinks/riemann/driver.go [hunk 6]
```diff
@@ -172,42 +179,33 @@ func (sink *riemannSink) ExportData(dataBatch *core.DataBatch) {
...
+	if len(events) > 0 {
+		sink.sendData(events)
 	}
 }
 
-func (sink *riemannSink) sendData(dataEvents []riemann_api.Event) {
+// Send Events to Riemann using the client from the sink.
+func (sink *riemannSink) sendData(events []riemanngo.Event) {
+	// do nothing of we are not connected
```
Comment (mcorbin) [inline metrics/sinks/riemann/driver.go]: Fixed, thanks

File: docs/sink-configuration.md [hunk 7]
```diff
@@ -135,16 +135,20 @@ For example,
...
 
 	--sink="riemann:<RIEMANN_SERVER_URL>[?<OPTIONS>]"
 
 The following options are available:
 
 * `ttl` - TTL for writes to Riemann. Default: `60 seconds`
-* `state` - FIXME. Default: `""`
-* `tags` - FIXME. Default. `none`
-* `storeEvents` - Control storage of events. Default: `true`
+* `state` - Default: `""`
```
Comment (mcorbin) [inline docs/sink-configuration.md]: I have added a description, thanks

File: metrics/sinks/riemann/driver_test.go (hunk 2 above)
Comment (mcorbin) [inline metrics/sinks/riemann/driver_test.go]: Fixed, thanks

File: metrics/sinks/riemann/driver_test.go (hunk 1 above)
Comment (mcorbin) [inline metrics/sinks/riemann/driver_test.go]: Fixed, thanks

File: metrics/sinks/riemann/driver_test.go [hunk 8]
```diff
@@ -175,23 +247,139 @@ func TestStoreMultipleDataInput(t *testing.T) {
...
+				Key:   pb.String("namespace_id"),
+				Value: pb.String("123"),
+			},
+			{
+				Key:   pb.String("pod_id"),
+				Value: pb.String("aaaa-bbbb-cccc-dddd"),
+			},
+		},
 	}
+	var expectedEvents = []*proto.Event{event1, event2, event3, event4, event5}
```
Comment (jsoriano) [inline metrics/sinks/riemann/driver_test.go]: I meant to define the events inline, not in the same line :) i.e:
```
expectedEvents := []*proto.Event{
    &proto.Event{
        ...
    },
    &proto.Event{
        ...
    },
    ...
}
```

Comment (mcorbin) [inline metrics/sinks/riemann/driver_test.go]: Fixed now ;)

**Comment by jsoriano** [comment]
LGTM

**Comment by DirectXMan12** [comment]
@k8s-bot ok to test

will merge once tests pass due to LGTM from sink-owner.

We're probably going to want a release note on the label changes @piosz

**Comment by DirectXMan12** [comment]
Thanks for this!

**Pull request merged**